core/Tracker/GoalManager.php

```diff
@@ -195,6 +207,42 @@ public function detectGoalMatch($goal, Action $action)
         return $action->getActionUrl();
     }
 
+    private function detectNumericGoalMatch($goal, Action $action, VisitProperties $visitProperties, Request $request)
+    {
+        switch ($goal['match_attribute']) {
+            case 'visit_duration':
+                $firstActionTime = $visitProperties->getProperty('visit_first_action_time');
+                if (empty($firstActionTime)) {
+                    return null;
+                }
+
+                $visitDurationInSecs = $request->getCurrentTimestamp() - ((int) $firstActionTime);
+                $valueToMatchAgainst = $visitDurationInSecs / 60;
+                break;
+            default:
+                return null;
+        }
+
+        $pattern = (int) $goal['pattern'];
+
+        Common::printDebug("Matching {$goal['match_attribute']} (current value = $valueToMatchAgainst, idGoal = {$goal['idgoal']}) {$goal['pattern_type']} $pattern.");
+
+        switch ($goal['pattern_type']) {
+            case '>':
```

Comment:
As mentioned in the other goal PR be good to use `greater_than` or something similar for consistency and readability.